docs/events/jarsigning.md

```diff
@@ -0,0 +1,83 @@
+Signing a JAR
+=============
+
+Forge allows Mods to sign their jar files with a signature. However, these signatures are not used
+as security and shouldn't be used this way. Signatures are used as sanity checks, so that Modders
+are able to check if he is running his own un-edited code.
+
+!!! note
+
+	If someone really wants to get around this system, they can. This system is not one hundred percent effective. This system is event based, meaning that an invalid key will cause an event to be executed.
+	
+Creating a keystore
+-------------------
+A **keystore** is a private database file which holds the required information to successfully sign the jar.
+To sign a jar, you need both a public and a private key. The public key will be later used to check if the
+jar is correctly signed.
+
+To generate a key, execute the following command and follow the instructions given by the keytool.
+The key should be **SHA-1 encoded**.
+```java
+keytool -genkey -alias signFiles -keystore examplestore.jks
+```
+The `keytool` is part of the Java Development Kit and can be found in the underlying `bin` directory.
+The `alias signFiles` indicates that the alias should be used in future to refer to the keystore entry and
+`-keystore examplestore.jks` means that the keystore will be saved to the file `examplestore.jks`.
+
+!!! note
+
+	A correct setup of the Java Development Kit is required!
```

Comment:
You don't need to mention this. The JDK is needed for anything related to programming Java.